src/buildercore/lifecycle.py

```diff
@@ -292,22 +283,43 @@ def _update_dns_a_record(zone_name, name, value):
     # "zone_name" => "elifesciences.org"
     # "name" => "foo--journal.elifesciences.org"
     # "value" => "1.2.3.4"
-    zone = _r53_connection().get_zone(zone_name)
-    a_record = zone.get_a(name)
-    if a_record:
-        if a_record.resource_records == [value]:
-            LOG.info("No need to update DNS record %s (already %s)", name, value)
-        else:
-            LOG.info("Updating DNS record %s to %s", name, value)
-            zone.update_a(name, value)
-    else:
+    r53 = _r53_connection()
+    result = r53.list_resource_record_sets(HostedZoneId=zone_name, StartRecordName=name, StartRecordType="A")
+    a_record_list = result['ResourceRecordSets']
+
+    # "zero or one 'A' records expected for 'foo--journal.elifesciences.org', found 2"
+    ensure(len(a_record_list) < 2, "zero or one 'A' records expected for %r, found %s" % (name, len(a_record_list)))
+    a_record = first(a_record_list)
+    if lookup(a_record, 'ResourceRecords.0.Value', None) == value:
+        # "No need to update DNS record 'foo--journal.elifesciences.org' (already 1.2.3.4)"
+        LOG.info("No need to update DNS record %r (already %s)", name, value)
+        return
+
+    if not a_record:
         # lsh@2021-08-02: record doesn't exist. This case almost never happens.
         # It *did* happen when another journal instance was brought up using the `prod` config.
         # It overwrote the DNS entries for `journal--prod` and then destroyed them when it rolled back.
         # `lifecycle.update_dns` is now the recommended way to fix broken DNS.
         LOG.warning("DNS record %s does not exist!", name)
         LOG.info("Creating DNS record %s with %s", name, value)
-        zone.add_a(name, value)
+    else:
+        LOG.info("Updating DNS record %s to %s", name, value)
+
+    r53.change_resource_record_sets(**{
+        "HostedZoneId": zone_name,
+        "ChangeBatch": {
+            "Changes": {
+                "Action": "UPSERT",
+                "ResourceRecordSet": {
+                    "Name": name,
+                    "Type": "A",
+                    "ResourceRecords": [
+                        {"Value": value}
+                    ]
+                }
+            }
+        }
+    })
```

Comment:
I don't think we have any test coverage here. I don't see any of the log messages during the integration tests. Can Moto help us here?